File: flutter_Backend/CreateApi/src/main/java/com/shohab/CreateApi/model/Bonus.java
package com.shohab.CreateApi.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
@Data
@NoArgsConstructor
@Entity
@AllArgsConstructor
public class Bonus {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Employee employee;
    private BigDecimal amount;
    private String bonusDate;

    public Bonus(Employee employee) {
        this.employee = employee;
    }


}

File: PayrollProjectNew/payroll_app/lib/src/features/bonus/model/bonus.dart
import 'package:payroll_app/src/features/employee/model/employee.dart';

class Bonus {
  int? id;
  Employee? employee;
  int? amount;
  String? bonusDate;

  Bonus({this.id, this.employee, this.amount, this.bonusDate});

  Bonus.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    employee = json['employee'] != null
        ? Employee.fromJson(json['employee'])
        : null;
    amount = json['amount'];
    bonusDate = json['bonusDate'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    if (this.employee != null) {
      data['employee'] = this.employee!.toJson();
    }
    data['amount'] = this.amount;
    data['bonusDate'] = this.bonusDate;
    return data;
  }
}